query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -35,14 +44,37 @@
   // while the selectedColumns
   // is a set of column names.
   private final LinkedHashSet<Expression> allSelections;
+  private final Duration timeSeriesPeriod;
 
   public ExecutionContext(String tenantId, QueryRequest request) {
     this.tenantId = tenantId;
     this.selectedColumns = new LinkedHashSet<>();
     this.allSelections = new LinkedHashSet<>();
+    this.timeSeriesPeriod = setTimeSeriesPeriod(request);
     analyze(request);
   }
 
+  private Duration setTimeSeriesPeriod(QueryRequest request) {
+    Duration period = Duration.ZERO;
```

Comment:
So if this is unset, we're returning a zero duration instead of an empty optional. On the other side, the getter checks for a zero and converts it back to an optional. We can cut out the middleman (and the implicit contract!) by return an optional, saving the optional into the variable, and making the getter a straight access.